### ISSUE TITLE
renderAtom doesn't scape '&'

### ISSUE THREAD

**Issue by treeman** [jaspervdj/hakyll#706]
I noticed that atom feeds no longer validate if you have a title with '&' in it. I think renderAtom should be able to escape the special chars.

Hakyll 4.12.5.1

**Comment by Tarektouati** [comment]
Can I work on this?

**Comment by jaspervdj** [comment]
@Tarektouati That would be awesome!  Let me know if you need any help.

**Comment by jaspervdj** [comment]
This also applies to RSS; I hit this myself recently: https://github.com/jaspervdj/jaspervdj/issues/20

**Comment by Tarektouati** [comment]
@jaspervdj  Cool! I'll open a PR to fix Atom and RSS  feed.
I'll ping you if I need help, thanks

**Comment by Tarektouati** [comment]
@jaspervdj  Hey ! is there any getting started for contributors/guide?

**Comment by jaspervdj** [comment]
@Tarektouati Unfortunately not!  It would be nice to put that together but I don't think I have bandwidth for that soon.

For this particular issue, I would look into the [Feed](https://github.com/jaspervdj/hakyll/blob/v4.13.0.1/lib/Hakyll/Web/Feed.hs) module.  In addition to that, `escapeHtml` from the [Html](https://github.com/jaspervdj/hakyll/blob/v4.13.0.1/lib/Hakyll/Web/Html.hs) module is probably useful as well.

**Comment by Tarektouati** [comment]
@jaspervdj thanks ! I’ll figure out how. Can you open an issue about ` introduce contribution/development guide`? I’ll work when I’m done with this issue

**Comment by jaspervdj** [comment]
So I briefly had a stab at this in #750 but as I finished the implementation I realised this isn't really what we want.

It's not really clear that whatever is in `title` should _not_ be HTML, and I assume that some people may actually put some HTML in there (e.g. `<strong>`) and then use that inside an `h1`.  They may or may not filter out these tags for Atom/RSS.  Implementing this escaping would break that.

I think it's safer for backwards-compatibility to do this fix on the client side by adding something like this to your code:
```
postCtx :: Tags -> Context String
postCtx tags = mconcat
    [ modificationTimeField "mtime" "%U"
    , dateField "date" "%B %e, %Y"
    , tagsField "tags" tags
    , Context $ \key -> case key of
        "title" -> unContext (mapContext escapeHtml defaultContext) key
        _       -> unContext mempty key
    , defaultContext
    ]
```

This seems necessary anyway if you want to use `&` in other places on your website -- although I suppose browsers care less about this than XML readers (I also had this bug on my website and hadn't noticed it).
